src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
+
+  static removeDeadAnimal() {
+    Animal.alive = Animal.alive.filter((item) => item.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+    super.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden && prey.health > 0) {
```

Comment:
[CODE PERFORMANCE] There is an unnecessary check for `prey.health > 0` in the `if` condition. The `bite` method should only check if the `prey` is an instance of `Herbivore` and is not hidden. The health check is redundant because the health is being reduced in the next line, and the removal of dead animals is already handled by `Animal.removeDeadAnimal`.